src/main/java/com/rethinkdb/net/DefaultConnectionFactory.java

```diff
@@ -75,18 +75,23 @@ SocketWrapper connect() {
 
                 // should we secure the connection?
                 if (sslContext != null) {
-                    SSLSocket sslSocket = (SSLSocket) sslContext.getSocketFactory().createSocket(
-                        socket,
-                        socket.getInetAddress().getHostAddress(),
-                        socket.getPort(),
-                        true);
+                    try (
```

Comment:
It seems the existing code just creates an SSLSocket and then lets it get GC'd without closing it, assigning inputStream and outputStream instead.  I guess the underlying socket does get closed later.

Right now I think we _don't_ want this `try` block which closes the sslSocket at the end of the block.  Why would we close() the sslSocket after constructing streams from it?  Wouldn't those streams stop working?

I think we'd want sslSocket to be assigned to a member variable such that its lifetime is held alive as long as the streams exist.

I'll need to investigate what's going on a little bit further before merging this PR.  So right now I'm leaving this open.